drivers/sensor/nxp/fxls8974/fxls8974.c

```diff
@@ -379,9 +379,7 @@ static int fxls8974_channel_get(const struct device *dev,
 
 			val += FXLS8974_MAX_ACCEL_CHANNELS;
 
-			if (fxls8974_get_temp_data(dev, val)) {
-				return -EIO;
-			}
+			return fxls8974_get_temp_data(dev, val);
```

Comment:
Please also change the return type from int to void.

Reply:
Hi @str4t0m,

Thanks for your feedback.

Regarding the return type change, I believe we should keep the return type of fxls8974_get_temp_data as int because the return type of the fxls8974_channel_get function is an integer. Since fxls8974_get_temp_data is called within fxls8974_channel_get, we need to return an integer value for error handling and consistency within the function.

This is just my perspective; I'd appreciate your thoughts on this.

Reply:
Ping @str4t0m - does your NACK still stand in the light of this reply?

Reply:
I don't understand, if you think this function should return an integer then why are you opposed to checking that return value

Reply:
In second glance, I think the description on the PR is misleading, it seems from the code you are just trying to return the value from that function, instead of translating it to -EIO. Which makes sense. It seems like it's not really because it always returns 0 that you're changing this

Reply:
Hi @decsny 

> In second glance, I think the description on the PR and commit message is misleading, it seems from the code you are just trying to return the value from that function, instead of translating it to -EIO. Which makes sense. It seems like it's not really because it always returns 0 that you're changing this

I believe the commit message aligns well with what you stated: 
"Since the function always returns zero, the error-checking if condition is redundant and can be removed."